src/coponents/ContactUsModal.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useState } from 'react';
+import {
+  Text,
+  View,
+  StyleSheet,
+  SafeAreaView,
+  TouchableOpacity,
+  TextInput,
+  Modal,
+} from 'react-native';
+import { Colors } from '../utils/colors';
+import { Dropdown } from 'react-native-element-dropdown';
+import { useMutation } from '@tanstack/react-query';
+import { submitContactUsForm } from '../utils/userService';
+import Toast from 'react-native-toast-message';
+import ModalHeader from './ModalHeader';
+
+type ContactUsModalProps = {
+  isVisible: boolean;
+  onClose: () => void;
+  userToken: string;
+  userEmail: string;
+};
+
+const ContactUsModal: React.FC<ContactUsModalProps> = ({
+  isVisible,
+  onClose,
+  userToken,
+  userEmail,
+}) => {
+  type FormData = {
+    email: string | undefined;
+    support_type: string;
+    message: string;
+  };
+
+  const items = [
+    {
+      label: 'Question',
+      value: 'Question',
+    },
+    {
+      label: 'Bug Report',
+      value: 'Bug Report',
+    },
+    {
+      label: 'Suggestion',
+      value: 'Suggestion',
+    },
+    {
+      label: 'Compliment',
+      value: 'Compliment',
+    },
+  ];
+
+  const [value, setValue] = useState('Question');
+  const [comment, setComment] = useState('');
+
+  const renderItem = (item) => {
+    return (
+      <View style={styles.item}>
+        <Text style={styles.textItem}>{item.label}</Text>
+      </View>
+    );
+  };
+
+  const submitFormMutation = useMutation({
+    mutationFn: (formData: FormData) => submitContactUsForm(userToken, formData),
+    onSuccess: () => {
+      console.log('Form submitted successfully');
+      onClose();
+    },
+    onError: (error) => {
+      console.error('Error submitting form: ', error);
+    },
+  });
+
+  const handleOnPress = () => {
+    const formData: FormData = {
+      email: userEmail,
+      support_type: value,
+      message: comment,
+    };
+    submitFormMutation.mutate(formData);
+    setComment('');
+    setValue('Question');
+    Toast.show({
+      type: 'success',
+      text1: 'Message successfully sent to the admin.',
+    });
+  };
+
```

Comment:
Can you please add KeyboardAvoidingView? You can check `SignIn` component for reference